asmrun/Makefile

```diff
@@ -15,159 +15,167 @@
-.PHONY: install-default
+libasmruni.$(A): $(IOBJS)
+	$(call MKLIB,$@, $^)
 
-ifeq "$(RUNTIMED)" "runtimed"
-install::
-	cp libasmrund.a $(INSTALL_LIBDIR)/libasmrund.a
-	cd $(INSTALL_LIBDIR); $(RANLIB) libasmrund.a
-endif
+libasmrunp.$(A): $(POBJS)
+	$(call MKLIB,$@, $^)
 
-ifeq "$(RUNTIMEI)" "true"
-install::
-	cp libasmruni.a $(INSTALL_LIBDIR)/libasmruni.a
-	cd $(INSTALL_LIBDIR); $(RANLIB) libasmruni.a
-endif
+libasmrun_pic.$(A): $(PICOBJS)
+	$(call MKLIB,$@, $^)
 
-ifeq "$(PROFILING)" "true"
-install::
-	cp libasmrunp.a $(INSTALL_LIBDIR)/libasmrunp.a
-endif
+libasmrun_shared.$(SO): $(PICOBJS)
+	$(MKDLL) -o $@ $^ $(NATIVECCLIBS)
 
-ifeq "$(SHARED)" "shared"
-install::
-	cp libasmrun_pic.a $(INSTALL_LIBDIR)/libasmrun_pic.a
-	cd $(INSTALL_LIBDIR); $(RANLIB) libasmrun_pic.a
-	cp libasmrun_shared.so $(INSTALL_LIBDIR)/libasmrun_shared.so
-endif
+.PHONY: install
+install:
+	cp $(TARGETS) "$(INSTALL_LIBDIR)"
 
 $(LINKEDFILES): %.c: ../byterun/%.c
-	ln -s ../byterun/$*.c $*.c
+	$(LN) $< $@
+#	$(LN) ../byterun/$*.c $*.c
 
-clean::
-	rm -f $(LINKEDFILES)
+%.d.$(O): %.c
+	$(CC) -c $(DFLAGS) $(OUTPUTOBJ)$@ $<
 
-%.d.o: %.c
-	$(CC) -c $(DFLAGS) -o $@ $<
+%.i.$(O) : %.c
+	$(CC) -c $(IFLAGS) $(OUTPUTOBJ)$@ $<
 
-%.i.o : %.c
-	$(CC) -c $(IFLAGS) -o $@ $<
+%.p.$(O): %.c
+	$(CC) -c $(PFLAGS) $(OUTPUTOBJ)$@ $<
 
-%.p.o: %.c
-	$(CC) -c $(PFLAGS) -o $@ $<
+%.pic.$(O): %.c
+	$(CC) -c $(PICFLAGS) $(OUTPUTOBJ)$@ $<
 
-%.pic.o: %.c
-	$(CC) -c $(PICFLAGS) -o $@ $<
+%.$(O): %.c
+	$(CC) $(CFLAGS) -c $<
 
 %.o: %.S
-	$(ASPP) -DSYS_$(SYSTEM) -DMODEL_$(MODEL) -o $@ $< || \
+	$(ASPP) $(ASPPFLAGS) -o $@ $< || \
 	{ echo "If your assembler produced syntax errors, it is probably";\
           echo "unhappy with the preprocessor. Check your assembler, or";\
           echo "try producing $*.o by hand.";\
           exit 2; }
 
 %.p.o: %.S
-	$(ASPP) -DSYS_$(SYSTEM) -DMODEL_$(MODEL) $(ASPPPROFFLAGS) -o $@ $<
+	$(ASPP) $(ASPPFLAGS) $(ASPPPROFFLAGS) -o $@ $<
 
 %.pic.o: %.S
-	$(ASPP) -DSYS_$(SYSTEM) -DMODEL_$(MODEL) $(SHAREDCCCOMPOPTS) -o $@ $<
-
-%.o: %.s
-	$(ASPP) -DSYS_$(SYSTEM) -o $@ $<
-
-%.p.o: %.s
-	$(ASPP) -DSYS_$(SYSTEM) $(ASPPPROFFLAGS) -o $@ $<
+	$(ASPP) $(ASPPFLAGS) $(SHAREDCCCOMPOPTS) -o $@ $<
 
-%.pic.o: %.s
-	$(ASPP) -DSYS_$(SYSTEM) $(SHAREDCCCOMPOPTS) -o $@ $<
+%.obj: %.asm
+	$(ASM) $@ $<
 
-clean::
-	rm -f *.o *.a *.so *~
+.PHONY: clean
+clean:
+	rm -f $(LINKEDFILES)
+	rm -f *.$(O) *.$(A) *.$(SO) *~
 
-depend: $(COBJS:.o=.c) ${LINKEDFILES}
+ifneq "$(TOOLCHAIN)" "msvc"
+.PHONY: depend
+depend: $(COBJS:.$(O)=.c) $(LINKEDFILES)
 	$(CC) -MM $(FLAGS) *.c > .depend
 	$(CC) -MM $(FLAGS) -DPROFILING *.c | sed -e 's/\.o/.p.o/' >> .depend
-	$(CC) -MM $(FLAGS) -DDEBUG *.c | sed -e 's/\.o/.d.o/' >> .depend
-	$(CC) -MM $(FLAGS) -DCAML_INSTR *.c | sed -e 's/\.o/.i.o/' >> .depend
+	$(CC) -MM $(FLAGS) -DDEBUG *.c | sed -e 's/\.o/.d.o)/' >> .depend
+	$(CC) -MM $(FLAGS) -DCAML_INSTR *.c | sed -e 's/\.o/.i.o' >> .depend
+endif
 
+ifeq "$(UNIX_OR_WIN32)" "win32"
```

Comment:
Does anyone know if MSVC is usable to generate the .depend files? If so it would be possible to drop this completely simply by using "$(O)" above.

I would at least change the test to be on $(TOOLCHAIN) instead and I would name the file '.depend.msvc' instead.